### PR TITLE
Feature/81_Integrate-PFM-middleware

### PR DESCRIPTION
**Pull request by InnerPeace080**
# Description

- Upgrade cometbft to 0.37.6, ibc-go to 7.4.0, gogoproto to 1.4.12
- Integrate `packetForward` middleware for ibc transfer.

Closes: #81

---

## Author Checklist

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.

I have...

- [x] tackled an existing issue or discussed with a team member
- [x] left instructions on how to review the changes
- [x] targeted the correct branch (see [PR Targeting](https://github.com/titantkx/titan/blob/develop/CONTRIBUTING.md#pr-targeting))

## Reviewers Checklist

**All** items are required.
Please add a note if the item is not applicable
and please add your handle next to the items reviewed
if you only reviewed selected items.

I have...

- [x] confirmed all author checklist items have been addressed
- [x] reviewed content
- [x] confirmed all CI checks have passed

### CODE REVIEW

File: tests/e2e/cmd/setup/basic/init.sh [hunk 1]
```diff
@@ -84,7 +86,7 @@ config="
 .app_state.bank.denom_metadata[0].denom_units[3].denom = \"tkx\" |
 .app_state.bank.denom_metadata[0].denom_units[3].exponent = 18
 "
-echo $(jq "$config" tmp/val1/.titand/config/genesis.json) > tmp/val1/.titand/config/genesis.json
+echo "$(jq "$config" tmp/val1/.titand/config/genesis.json)" >tmp/val1/.titand/config/genesis.json
```
Comment (hugjobk) [inline tests/e2e/cmd/setup/basic/init.sh:89]:
```suggestion
echo $(jq "$config" tmp/val1/.titand/config/genesis.json) >tmp/val1/.titand/config/genesis.json
```

Comment (InnerPeace080) [reply tests/e2e/cmd/setup/basic/init.sh:89]: Wrap param string inside "" character can help avoid wrong echo output if string contain space or newline charater.

EX:

file `a.txt` with content:
```
a b
c
```

with command :

```shell
echo $(cat a.txt) > b.txt
```

will lead to `b.txt` content like this:

```
a b c
```

File: tests/e2e/cmd/setup/upgrade/setup.go [hunk 2]
```diff
@@ -12,11 +12,15 @@ import (
...
-const UpgradeName = "v2_0_1"
+const UpgradeName = "v3_0_0_rc_0"
 
 func Setup(m *testing.M, rootDir string, logger io.Writer) {
 	t := testutil.NewMockTest(os.Stderr)
 	defer t.Finish()
+	t.HandleOSInterrupt(func() {
+		defer setup.StopChain(t, logger, "docker-compose-genesis.yml")
+		defer setup.StopChain(t, logger, "docker-compose-local.yml")
+	})
```
Comment (hugjobk) [inline tests/e2e/cmd/setup/upgrade/setup.go:23]:
```suggestion

	// Set up signal handling for Ctrl+C
	sigChan := make(chan os.Signal, 1)
	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)

	go func() {
		fmt.Println("Received interrupt signal")
		setup.StopChain(t, logger, "docker-compose-genesis.yml")
		setup.StopChain(t, logger, "docker-compose-local.yml")
		os.Exit(1)
	}()
```

Comment (hugjobk) [reply tests/e2e/cmd/setup/upgrade/setup.go:23]: Don't need to cleanup tests on interrupt signal

File: tests/e2e/cmd/setup/basic/init.sh (hunk 1 above)
Comment (hugjobk) [reply tests/e2e/cmd/setup/basic/init.sh:89]: Then please add "" for similar echo commands as well

File: tests/e2e/cmd/setup/upgrade/setup.go (hunk 2 above)
Comment (InnerPeace080) [reply tests/e2e/cmd/setup/upgrade/setup.go]: I moved `HandleOSInterrupt` to `testutil/os`

File: tests/e2e/cmd/setup/upgrade-from-genesis/setup.go [hunk 3]
```diff
@@ -12,11 +12,15 @@ import (
...
 
-const UpgradeName = "v2_0_1"
+const UpgradeName = "v3_0_0_rc_0"
 
 func Setup(_ *testing.M, rootDir string, logger io.Writer) {
 	t := testutil.NewMockTest(os.Stderr)
 	defer t.Finish()
+	testutil.HandleOSInterrupt(func() {
+		defer setup.StopChain(t, logger, "docker-compose-genesis.yml")
+		defer setup.StopChain(t, logger, "docker-compose-local.yml")
```
Comment (hugjobk) [inline tests/e2e/cmd/setup/upgrade-from-genesis/setup.go:22]:
```suggestion
		setup.StopChain(t, logger, "docker-compose-genesis.yml")
		setup.StopChain(t, logger, "docker-compose-local.yml")
```

File: tests/e2e/cmd/setup/upgrade/setup.go [hunk 4]
```diff
@@ -12,11 +12,15 @@ import (
...
 
-const UpgradeName = "v2_0_1"
+const UpgradeName = "v3_0_0_rc_0"
 
 func Setup(m *testing.M, rootDir string, logger io.Writer) {
 	t := testutil.NewMockTest(os.Stderr)
 	defer t.Finish()
+	testutil.HandleOSInterrupt(func() {
+		defer setup.StopChain(t, logger, "docker-compose-genesis.yml")
+		defer setup.StopChain(t, logger, "docker-compose-local.yml")
```
Comment (hugjobk) [inline tests/e2e/cmd/setup/upgrade/setup.go:22]:
```suggestion
		setup.StopChain(t, logger, "docker-compose-genesis.yml")
		setup.StopChain(t, logger, "docker-compose-local.yml")
```

File: testutil/os.go [hunk 5]
```diff
@@ -16,3 +19,15 @@ func MustCreateTemp(t TestingT, name string) *os.File {
...
 	return file
 }
+
+func HandleOSInterrupt(f func()) {
+	// Set up signal handling for Ctrl+C
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-sigChan
+		fmt.Println("Received interrupt signal, cleaning up...")
```
Comment (hugjobk) [inline testutil/os.go:29]:
```suggestion
		fmt.Println("Received interrupt signal")
```

**Review by hugjobk — APPROVED** [review]
LGTM!

**Pull request merged**